Realm/Tests/RealmTests.mm

```diff
@@ -1423,21 +1423,111 @@ - (void)testIsEmpty {
     XCTAssertFalse(realm.isEmpty, @"Realm should not be empty after committing a write transaction that added an object.");
 }
 
+- (void)testRealmFileAccessNilPath {
+    XCTAssertThrows([RLMRealm realmWithURL:self.nonLiteralNil], @"nil path");
+}
 
-- (void)testRealmFileAccess
+- (void)testRealmFileAccessNoExistingFile
 {
-    XCTAssertThrows([RLMRealm realmWithURL:self.nonLiteralNil], @"nil path");
+    NSURL *fileURL = [NSURL fileURLWithPath:RLMRealmPathForFile(@"filename.realm")];
+    [[NSFileManager defaultManager] removeItemAtPath:fileURL.path error:nil];
+    assert(![[NSFileManager defaultManager] fileExistsAtPath:fileURL.path]);
 
+    NSError *error;
+    RLMRealmConfiguration *configuration = [RLMRealmConfiguration defaultConfiguration];
+    configuration.fileURL = fileURL;
+    XCTAssertNotNil([RLMRealm realmWithConfiguration:configuration error:&error],
+                    @"Database should have been created");
+}
+
+- (void)testRealmFileAccessInvalidFile
+{
     NSString *content = @"Some content";
     NSData *fileContents = [content dataUsingEncoding:NSUTF8StringEncoding];
     NSURL *fileURL = [NSURL fileURLWithPath:RLMRealmPathForFile(@"filename.realm")];
+    [[NSFileManager defaultManager] removeItemAtPath:fileURL.path error:nil];
+    assert(![[NSFileManager defaultManager] fileExistsAtPath:fileURL.path]);
     [[NSFileManager defaultManager] createFileAtPath:fileURL.path contents:fileContents attributes:nil];
 
     NSError *error;
     RLMRealmConfiguration *configuration = [RLMRealmConfiguration defaultConfiguration];
     configuration.fileURL = fileURL;
     XCTAssertNil([RLMRealm realmWithConfiguration:configuration error:&error], @"Invalid database");
     XCTAssertNotNil(error, @"Should populate error object");
+    // Validate error contents
+    XCTAssertEqualObjects(error.domain, RLMErrorDomain);
+    XCTAssertEqual(error.code, RLMErrorFileAccess);
+    XCTAssert([error.userInfo[@"Underlying"] rangeOfString:@"Realm file has bad size"].location != NSNotFound);
+}
+
+- (void)testRealmFileAccessBadEncryptionKey
```

Comment:
There's a family of tests in EncryptionTests which cover this (testReopenWithNoKeyThrows, testReopenWithWrongKeyThrows, testOpenUnencryptedWithKeyThrows), so rather than duplicating the testing it'd be better to just make those tests actually check what exception is thrown.